### ISSUE TITLE
emu.setregister is broken in mGBA

### ISSUE THREAD

**Issue by RetroEdit** [TASVideos/BizHawk#1982]
Someone mentioned they were having some trouble with it related to a GBA Pokemon game on Discord, and I did a small test that confirmed it has problems in at least some cases.

----

What I tested is as follows: using BizHawk 2.4.1 and Final Fantasy I and II: Dawn of Souls (USA, Australia) (`SHA1: 6472695D69661490F78245E2982E1E676C080BE7, MD5: 5D29999685413C4D2BEC10D3160F6EE6`), I first started by running the following script:

```lua
local function fifteen_puzzle_seed()
    local reg = emu.getregisters()
    local r0, r1 = reg["R0"], reg["R1"]
    console.log(string.format("RNG: %8X, VCOUNT: %2X", r0, r1))
    emu.setregister("R1", r1 + 1)
end

event.onmemoryexecute(fifteen_puzzle_seed, 0x0803C23E, "15_puzzle_seed")
```

I then loaded a savestate immediately prior to when the 15-puzzle minigame was being loaded, which triggered this particular execution callback. When this script is loaded, the `getregister` portion of the script and outputs the numbers to the console. The `setregister` fails and outputs `Error message: A .NET exception occured in user-code`. For convenience, I have included this savestate and script in an attachment: [BizHawk setregister bug.zip](https://github.com/TASVideos/BizHawk/files/4582244/BizHawk.setregister.bug.zip)

----

I include this description simply for the sake of documenting the problem in a way that I and hopefully others can reproduce. It likely occurs in more general cases, but I don't have the time at the moment to test them. My script also has the disadvantage of using execute callbacks, but this seems to be the main context in which register getting and setting would be used anyway.

**Comment by YoshiRulz** [comment]
Is this a regression from 2.4?

**Comment by adelikat** [comment]
It can't be a regression because I added register setting in 2.4.1 :)

**Comment by RetroEdit** [comment]
Looks like adelikat's new code worked fine, but it just never ended up in the mGBA submodule included by BizHawk. Fixed in d37de2e5.
